### DISCUSSION TITLE
Use ramdisk on Pi 4B for 3rd etcd node

### DISCUSSION THREAD

**Discussion by drpump** [k3s-io/k3s#9989] [Q&A]
Hi, I am currently running k3s with a single master node and SQLite DB. I would like to move to a higher availability configuration using etcd. The master node has a fast NVMe drive. I have a second, similarly fast node. I also have a spare RPi 4B with 4GB RAM but am aware that etcd needs fast storage (i.e. not the SD card). 

My thinking is that since my state DB is small (currently ~50MB) I could create a 500MB or 1GB RAM disk on the RPi and mount it on `/var/lib/rancher/k3s/server/db/etcd`, making it fast enough to operate as an etcd master node. This avoids the cost of adding a third node with fast NVMe, which I don’t really need for any other reason at present (my two nodes have plenty of capacity). 

Is there any reason not to do this? Will etcd automatically repopulate the DB state after a restart on the RPi?

**Comment by djw4** [comment]
I’m in the same position where I have a cluster board with SBCs that are (currently at least) quite expensive. I also have a spare RPi4 that I am no longer using, so I was thinking to repurpose it in an identical manner but wary of performance degradation.
